hackathon_site/dashboard/frontend/src/slices/event/teamDetailSlice.ts

```diff
@@ -30,6 +30,46 @@ interface RejectValue {
     message: any;
 }
 
+interface idProvidedParameters {
+    profileId: number;
+    idProvided: boolean;
+    attended: boolean;
+}
+
+export const updateParticipantIdProvided = createAsyncThunk<
+    Profile,
+    idProvidedParameters,
+    { state: RootState; rejectValue: RejectValue; dispatch: AppDispatch }
+>(
+    `${teamDetailReducerName}/updateParticipantIdProvided`,
+    async ({ profileId, idProvided, attended }, { rejectWithValue, dispatch }) => {
+        try {
+            const response = await patch<Profile>(`/api/event/profiles/${profileId}/`, {
+                id_provided: !idProvided,
+                attended: attended,
+            });
+            return response.data;
+        } catch (e: any) {
+            const message =
+                e.response.statusText === "Not Found"
+                    ? `Could not find participant: Error ${e.response.status}`
```

Comment:
error message suggestion:
`Could not update participant id status: Error {status code}`

Reply:
changed